dd-sdk-android/src/main/kotlin/com/datadog/android/rum/internal/domain/RumEventData.kt

```diff
@@ -23,8 +23,46 @@ internal sealed class RumEventData(val category: String) {
     internal data class View(
         val name: String,
         val durationNanoSeconds: Long,
+        val measures: Measures = Measures(),
         val version: Int = 1
-    ) : RumEventData("view")
+    ) : RumEventData("view") {
+
+        fun incrementErrorCount(): View {
+            return copy(
+                measures = measures.incrementErrorCount()
+            )
+        }
+
+        fun incrementResourceCount(): View {
+            return copy(
+                measures = measures.incrementResourceCount()
+            )
+        }
+
+        fun incrementUserActionCount(): View {
+            return copy(
+                measures = measures.incrementUserActionCount()
+            )
+        }
+
+        internal data class Measures(
+            val errorCount: Int = 0,
```

Comment:
do you think we should make these Atomic ? I am thinking that if one of these *startResource* or *startUserAction* calls will be made from different threads you will have wrong incrementation here.

Reply:
Nope, every call to the RumMonitor are already `synchronized`, which is the only way to change these values.